src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/serviceprovider/performance/PerformanceReportProcessor.kt

```diff
@@ -44,6 +43,12 @@ class PerformanceReportProcessor(
       endRequestedReason = referral.endRequestedReason?.description,
       eosrSubmittedAt = referral.endOfServiceReport?.submittedAt,
       concludedAt = referral.concludedAt,
+      referralEndState = referral.endState,
+      dateSupplierAssessmentBooked = saa?.firstAppointment?.createdAt,
+      dateSupplierAssessmentFirstScheduledFor = saa?.firstAppointment?.appointmentTime,
+      dateSupplierAssessmentFirstNotAttended = saa?.firstAppointmentWithNonAttendance?.appointmentTime,
+      dateSupplierAssessmentFirstAttended = saa?.firstAttendedAppointment?.appointmentTime,
+      supplierAssessmentAttendedOnTime = saa?.firstAttendedAppointment?.attended == Attended.YES,
```

Comment:
This would be true if someone did not attend -- in that case we should use `null` here

Can we add a test for that?

Reply:
Oh... This is `firstAttended...` so this won't happen. Never mind me!

Reply:
no actually i think you're right. `saa?.firstAttendedAppointment?.attended` resolves to `null` and `null == ...` is false. good spot.